### PR TITLE
Fixes the BSA asking for the target twice

### PR DESCRIPTION
**Pull request by Arrow768**

### CODE REVIEW

File: code/modules/awaymissions/bluespaceartillery.dm [hunk 1]
```diff
@@ -44,20 +44,13 @@
 	if(!istype(t))
 		return
 	if ((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
-		var/A
-		A = input("Area to jump bombard", "Open Fire", A) in teleportlocs
-		var/area/thearea = teleportlocs[A]
 		if (usr.stat || usr.restrained()) return
```
Comment (skull132) [inline code/modules/awaymissions/bluespaceartillery.dm]:
```suggestion
		if (user.stat || user.restrained()) return
```

while we're here

File: code/modules/awaymissions/bluespaceartillery.dm [hunk 2]
```diff
@@ -44,20 +44,13 @@
 	if(!istype(t))
 		return
 	if ((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
-		var/A
-		A = input("Area to jump bombard", "Open Fire", A) in teleportlocs
-		var/area/thearea = teleportlocs[A]
 		if (usr.stat || usr.restrained()) return
-		if(src.reload < 180) return
+		if (src.reload < 180) return
 		if ((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
```
Comment (skull132) [inline code/modules/awaymissions/bluespaceartillery.dm]:
```suggestion
		if ((user.contents.Find(src) || (in_range(src, user) && istype(src.loc, /turf))) || (istype(user, /mob/living/silicon)))
```

File: code/modules/awaymissions/bluespaceartillery.dm [hunk 3]
```diff
@@ -44,20 +44,13 @@
 	if(!istype(t))
 		return
 	if ((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
```
Comment (skull132) [inline code/modules/awaymissions/bluespaceartillery.dm]:
```suggestion
	if ((user.contents.Find(src) || (in_range(src, user) && istype(src.loc, /turf))) || (istype(user, /mob/living/silicon)))
```

File: code/modules/awaymissions/bluespaceartillery.dm [hunk 4]
```diff
@@ -43,21 +43,14 @@
...
+		if (user.stat || user.restrained()) return
+		if (src.reload < 180) return
+		if ((user.contents.Find(src) || (in_range(src, user) && istype(src.loc, /turf))) || (istype(user, /mob/living/silicon)))
 			command_announcement.Announce("Bluespace artillery fire detected. Brace for impact.")
 			to_world(sound('sound/effects/yamato_fire.ogg'))
 			message_admins("[key_name_admin(usr)] has launched an artillery strike.", 1)
-			var/list/L = list()
-			for(var/turf/T in get_area_turfs(thearea.type))
-				L+=T
-			var/loc = pick(L)
```
Comment (Sindorman) [inline code/modules/awaymissions/bluespaceartillery.dm]: Wouldn't this make BSA accurate? No off-set. If it does then we need to Changelog it.

Comment (Arrow768) [reply code/modules/awaymissions/bluespaceartillery.dm]: No. Check lines 26 to 41

**Pull request merged**